src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDeliveryService.cs

```diff
@@ -44,5 +53,100 @@ public PurchaseOrderDeliveryService(IRepository<PurchaseOrderDelivery, PurchaseO
 
             return result.OrderBy(x => x.OrderNumber);
         }
+
+        public PurchaseOrderDelivery UpdateDelivery(
+            PurchaseOrderDeliveryKey key,
+            PurchaseOrderDelivery from,
+            PurchaseOrderDelivery to,
+            IEnumerable<string> privileges)
+        {
+            if (!this.authService.HasPermissionFor(AuthorisedAction.PurchaseOrderUpdate, privileges))
+            {
+                throw new UnauthorisedActionException("You are not authorised to acknowledge orders.");
+            }
+
+            var entity = this.repository.FindById(key);
```

Comment:
I think you've already done this findById in the facade service and actually update these fields there too after this - do you want to pass the entity in from there and update them by reference here sort of thing? Probs at least stop duplicate these field them unless I've read it wrong

Reply:
I just forgot to delete the field updates from the facade when I refactored the update into the domain. Good spot!

Reply:
Ah yeah makes sense. There's still the 2 findById calls - do you wanna pass the entity from the facade to the domain or happy it'll be updated by reference/ef magic by the time it gets send to the resource builder?

Reply:
No, I can completely remove any persistence dependencies from my facade now since it's all done in the domain